src/treetensornetworks/projttns/projttnsum.jl

```diff
@@ -1,59 +1,51 @@
 """
 ProjTTNSum
 """
-struct ProjTTNSum{V}
-  pm::Vector{ProjTTN{V}}
-  function ProjTTNSum(pm::Vector{ProjTTN{V}}) where {V}
-    return new{V}(pm)
+struct ProjTTNSum{V} <: AbstractProjTTN{V}
+  terms::Vector{T} where {T<:AbstractProjTTN{V}}
+  function ProjTTNSum(terms::Vector{T}) where {V,T<:AbstractProjTTN{V}}
+    return new{V}(terms)
   end
 end
```

Comment:
```suggestion
vertextype(::Type{<:AbstractProjTTN{V}}) where {V} = V
vertextype(p::AbstractProjTTN) = typeof(p)

struct ProjTTNSum{V,T<:AbstractProjTTN{V}} <: AbstractProjTTN{V}
  terms::Vector{T}
  function ProjTTNSum(terms::Vector{<:AbstractProjTTN})
    return new{vertextype(eltype(terms)),eltype(terms)}(terms)
  end
end
```
What you wrote would still be seen as a field with an abstract type by the Julia compiler. The way I wrote it, if `T` is a concrete type, then `ProjTTNSum` will be a concrete type. See https://docs.julialang.org/en/v1/manual/performance-tips/#Avoid-fields-with-abstract-type.

I see part of the problem you were having was the details of how to handle the type parameter `V` of the `AbstractProjTTN`, dealing with nested type parameters like that can be tricky. I think what I wrote above is correct, or at least close to what we should do.

Reply:
Note that `vertextype` is a function defined in `NamedGraphs.jl` so we should overload it from there.